test/state_machine/stm_tests.cpp

```diff
@@ -12,9 +15,26 @@ void testTransition(std::unique_ptr<state_machine::StateMachine> &stm,
   ASSERT_TRUE(stm->getCurrentState() == expected_state);
 }
 
+std::shared_ptr<core::Mqtt> getMqtt()
```

Comment:
unless you have a way to use the broker as part of the tests, you're gonna want an `IMqtt` interface and a `MockMqtt` class in `lib/utils` I reckon